File: q0150/q0150.java
class Solution {
    public int evalRPN(String[] tokens) {
        Deque<Integer> stack = new ArrayDeque<Integer>();
        for (String s: tokens) {
            // If we encounter an operator, then the two most immediate integer on the stack are the left and right operands
            if (s.equals("+") || s.equals("-") || s.equals("*") || s.equals("/")) {
                int r = stack.pop();
                int l = stack.pop();
                if (s.equals("+")) stack.push(l + r);
                if (s.equals("-")) stack.push(l - r);
                if (s.equals("*")) stack.push(l * r);
                if (s.equals("/")) stack.push(l / r);
            } else {
                stack.push(Integer.parseInt(s));
            }
        }
        return stack.peek();
    }
}

File: q0150/q0150.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operandStack = []
        for token in tokens: 
            if token not in ['+', '-', '*', '/']:
                operandStack.append(int(token))
            else: 
                operand2 = operandStack.pop()
                operand1 = operandStack.pop()
                if token == '+': operandStack.append(operand1 + operand2)
                elif token == '-': operandStack.append(operand1 - operand2)
                elif token == '*': operandStack.append(operand1 * operand2)
                else: operandStack.append(int(operand1 / operand2))
        return operandStack.pop()